src/main/java/com/google/sps/servlets/UserLoginServlet.java

```diff
@@ -15,6 +17,16 @@
     urlPatterns = "/userapi")
 public class UserLoginServlet extends HttpServlet {
 
+  private SessionContext sessionContext;
+
+  public UserLoginServlet() {
+    this(SessionContext.getInstance());
+  }
+   // test only call the second one
+  public UserLoginServlet(SessionContext var){
```

Comment:
var is a reserved keyword in Java. Consider choosing variable names that are easier for readers to understand.

Reply:
Done